### PR TITLE
Resolve graphql calls app side

### PR DESCRIPTION
**Pull request by diegomura**
@tpreusse  Proof of concept of the idea you proposed about resolving graphql queries app side.

**What this does**
- Listens for graphQL messages
- Resolve call
- Returns response back to the webview apollo link

**What this doesn't (yet at least)**
- Subscriptions
- Using cookies on request

**Example**
Here you can see the phrase being returned to the webview using this new architecture.
It's also working for getting the feed, but not sure yet if that's because some old cookies present in  the cookie store

kapture 2018-06-08 at 13 10 10

### CODE REVIEW

**Comment by diegomura** [comment]
@tpreusse subscriptions ready, finally. Ready for review 😄

File: src/apollo/link.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const hasSubscriptionOperation = ({ query }) => (
+  query.definitions.some(
+    ({ kind, operation }) =>
+      kind === 'OperationDefinition' && operation === 'subscription'
+  )
+)
+
+export const httpLink = new HttpLink({ uri: API_URL })
+
+export const wsLink = new WebSocketLink({
```
Comment (tpreusse) [inline src/apollo/link.js]: do not export unless used

File: src/apollo/link.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { API_URL, API_WS_URL } from '../constants'
+
+const hasSubscriptionOperation = ({ query }) => (
+  query.definitions.some(
+    ({ kind, operation }) =>
+      kind === 'OperationDefinition' && operation === 'subscription'
+  )
+)
+
+export const httpLink = new HttpLink({ uri: API_URL })
```
Comment (tpreusse) [inline src/apollo/link.js]: IMO use the split link in the app `withApollo` too. And not export the http only one

File: src/screens/Web.js [hunk 3]
```diff
@@ -101,7 +140,7 @@ class Web extends Component {
         />
         <WebView
           ref={node => { this.webview = node }}
-          source={{uri: data.url}}
+          source={{ uri: DISCUSSIONS_URL }}
```
[the remark this answers is not in the document]
Comment (diegomura) [reply src/screens/Web.js]: Yep. Sorry. Don't know why but the old `data.url` changed and couldn't identify the cause. fixing this now

**Pull request merged**